test/functional/Valhalla/src_qtypes/org/openj9/test/lworld/ValueTypeGenerator.java

```diff
@@ -339,13 +345,58 @@ private static void generateFieldMethods(ClassWriter cw, String[] nameAndSigValu
 		}
 	}
 
-	private static void addInit(ClassWriter cw) {
+	private static void addInit(ClassWriter cw, String className, String makeValueSig, String[] fields, boolean[] isStrictInitField) {
 		MethodVisitor mv = cw.visitMethod(ACC_PUBLIC, "<init>", "()V", null, null);
 		mv.visitCode();
+		boolean needsWideStack = false;
+		for (int i = 0; i < fields.length; i++) {
+			if (!isStrictInitField[i]) continue;
+			String[] nameAndSigValue = fields[i].split(":");
+			String sig = nameAndSigValue[1];
+			mv.visitVarInsn(ALOAD, 0);
+			switch (sig.charAt(0)) {
+				case 'J':
+					mv.visitInsn(LCONST_0);
+					needsWideStack = true;
+					break;
+				case 'D':
+					mv.visitInsn(DCONST_0);
+					needsWideStack = true;
+					break;
+				case 'F':
+					mv.visitInsn(FCONST_0);
+					break;
+				case 'I':
+				case 'S':
+				case 'B':
+				case 'C':
+				case 'Z':
+					mv.visitInsn(ICONST_0);
+					break;
+				default:
+					if (sig.charAt(0) == 'L') {
```

Comment:
The first character will always be 'L'. Can you remove this and other similar comparisons in this file? They are likely a relic from when `Q` was a possible value.